src/core/sys/posix/sys/time.d

```diff
@@ -28,7 +28,55 @@ else version (WatchOS)
     version = Darwin;
 
 version (Posix):
-extern (C) nothrow @nogc:
+nothrow @nogc:
+
+
+/* macros provided to operate on timeval structures
+ * they are extern(D) because they are not really C symbols, just macros
+ */
+
+pure @safe void timeradd(const timeval* a, const timeval* b,
+        timeval* result)
+{
+    result.tv_sec = a.tv_sec + b.tv_sec;
+    result.tv_usec = a.tv_usec + b.tv_usec;
```

Comment:
> Well, it makes sense, just a weird sense, just the same as having usecs > 1_000_000

Right, not how one might expect! For instance, if you had:

```D
auto ts1 = timeval(1, 5_000_000);
auto ts2 = timeval(5, 1_000_000);
timeval ts3;
timer_add(&ts1, &ts2, &ts3);
// a reasonable expectation:
assert(ts3.tv_sec == 12 && ts3.tv_usec == 0);
```
Thinking about it a bit, it's also possible that the current code works on timevals where the individual usecs might be above 1 million, but added together are still under 2 million, which would result in a valid timeval (ironically!).

> OK, I have to rethink this, and probably move it to the linux/bsd specific bindings...

I'm assuming this is an aid to porting C code. But how bad would it be to change the lines that use this function to instead call a more d-ish function?

In any case, even if the timer math functions are not posix, timeval is, and this function would be valid on all platforms that have the type.